File: examples/gateway-router.dart
import 'dart:math';

// OpenClaw gateway router — upstream node selection (port of nginx-gateway)
const nodes = ['gateway1.openclaw.internal', 'gateway2.openclaw.internal'];

String getNode() => nodes[Random().nextInt(nodes.length)];

void main() {
  print('OpenClaw routing to: ${getNode()}');
}

File: examples/gateway-router.ts
// OpenClaw gateway router — upstream node selection (port of nginx-gateway)
const nodes: string[] = [
  "gateway1.openclaw.internal",
  "gateway2.openclaw.internal",
];

function getNode(): string {
  return nodes[Math.floor(Math.random() * nodes.length)];
}

const target = getNode();
console.log(`OpenClaw routing to: ${target}`);
